### PR TITLE
Fixes #4171 Exclude judge.me script from delay JS

### LINKED ISSUE #4171: Compatibility with Judge.me Reviews when default exclusion are used
**Before submitting an issue please check that you’ve completed the following steps:**
- [x] Made sure you’re on the latest version
- [x] Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
We need to exclude an inline script from Judge.me Reviews to be delayed when the default exclusion list is used to avoid an issue.

**To Reproduce**
- Install Judge.me Product Reviews for WooCommerce plugin
- Activate Minify JS
- Activate Delay JS + include our default exclusion list
- See error

**Expected behavior**
Automatically exclude this pattern: `window.jdgmSettings`

**Additional context**
Website with Judge.me:
https://grillbillbbq.com/xl-kamado-bbq-23-inch-grill-bill-pro-ii-sr

The full inline script:

```
<script class='jdgm-settings-script'>window.jdgmSettings={"pagination":5,"badge_no_review_text":"0 reviews","badge_n_reviews_text":"{{ n }} Review/Reviews","badge_star_color":"#ffde00","hide_badge_preview_if_no_reviews":true,"widget_title":"Echte klantreviews","widget_open_form_text":"Scrhijf jouw review","widget_close_form_text":"Cancel","widget_refresh_page_text":"Ververs pagina","widget_summary_text":"Gebaseerd op {{ number_of_reviews }} review/reviews","widget_no_review_text":"Nog geen reviews","widget_name_field_text":"Naam","widget_verified_name_field_text":"Geverifieerde naam (public)","widget_name_placeholder_text":"Voer je naam in (public)","widget_required_field_error_text":"Dit veld is verplicht","widget_email_field_text":"E-mail","widget_verified_email_field_text":"Geverifieerd e-mailadres (privé, kan niet worden bewerkt)","widget_email_placeholder_text":"Voer je e-mailadres in (privé)","widget_email_field_error_text":"Gelieve een geldig e-mailadres in te geven.","widget_rating_field_text":"Review","widget_review_title_field_text":"Review Titel","widget_review_title_placeholder_text":"Geef je review een titel","widget_review_body_field_text":"Review","widget_review_body_placeholder_text":"Geef jouw mening","widget_pictures_field_text":"Foto's of video (optie)","widget_submit_review_text":"Verstuur review","widget_submit_verified_review_text":"Verstuur review","widget_submit_success_msg_with_auto_publish":"Bedankt voor je review! Als je je pagina ververst kun je jouw vers geschreven review teruglezen. Je kunt 'm zelfs delen via social media.","widget_submit_success_msg_no_auto_publish":"Bedankt voor je review! Je review word eerst beoordeeld. Je kunt 'm later ook delen via social media.","widget_show_default_reviews_out_of_total_text":"{{ n_reviews_shown }} wordt weergegeven bij {{ n_reviews }} beoordelingen.","widget_show_all_link_text":"Toon alles","widget_show_less_link_text":"Laat minder zien","widget_author_said_text":"{{ reviewer_name }} zei:","widget_days_text":"{{ n }} dagen geleden","widget_weeks_text":"{{ n }} week/weken geleden","widget_months_text":"{{ n }} maanden/maanden geleden","widget_years_text":"{{ n }} jaar/jaar geleden","widget_yesterday_text":"Gisteren","widget_today_text":"Vandaag","widget_replied_text":"Reactie van {{ shop_name }}:","widget_read_more_text":"Lees meer","widget_reviewer_name_as_initial":"last_initial","widget_rating_filter_see_all_text":"Alle beoordelingen bekijken","widget_sorting_most_recent_text":"Meest recent","widget_sorting_highest_rating_text":"Hoogste beoordeling","widget_sorting_lowest_rating_text":"Laagste beoordeling","widget_sorting_with_pictures_text":"Alleen foto's","widget_sorting_most_helpful_text":"Meest behulpzaam","widget_open_question_form_text":"Stel een vraag","widget_reviews_subtab_text":"Review","widget_questions_subtab_text":"Vragen","widget_question_label_text":"Vragen","widget_answer_label_text":"Antwoorden","widget_question_placeholder_text":"Schrijf je vraag","widget_submit_question_text":"Verstuur je vraag","widget_question_submit_success_text":"Bedankt voor je vraag! We zullen u op de hoogte stellen en uw vraag zal worden gepubliceerd zodra deze is beantwoord.","widget_star_color":"#ffd600","verified_badge_text":"Geverifieerd ","verified_badge_placement":"left-of-reviewer-name","widget_review_max_height":4,"widget_social_share":true,"widget_thumb":true,"widget_review_location_show":true,"widget_location_format":"country_iso_code","all_reviews_include_out_of_store_products":true,"all_reviews_out_of_store_text":"(uit winkel)","all_reviews_product_name_prefix_text":"over","enable_review_pictures":true,"enable_question_anwser":true,"review_date_format":"dd/mm/yyyy","default_sort_method":"most-helpful","widget_product_reviews_subtab_text":"Product recensies","widget_shop_reviews_subtab_text":"Winkel beoordelingen","show_product_url_for_grouped_product":true,"widget_sorting_pictures_first_text":"Foto's eerst","show_pictures_on_all_rev_page_mobile":true,"show_pictures_on_all_rev_page_desktop":true,"floating_tab_button_name":"★ Judge.me beoordelingen","floating_tab_title":"Laat klanten voor ons spreken","all_reviews_text_badge_text":"Klanten beoordelen ons {{ shop.metafields.judgeme.all_reviews_rating | round: 1 }}/5 gebaseerd op {{ shop.metafields.judgeme.all_reviews_count }} - recensies.","featured_carousel_title":"Laat klanten voor ons spreken","featured_carousel_count_text":"van {{ n }} recensies","featured_carousel_autoslide_interval":0,"featured_carousel_image_size":300,"featured_carousel_arrow_color":"#575757","picture_reminder_submit_button":"Upload afbeeldingen","enable_review_videos":true,"mute_video_by_default":true,"widget_sorting_videos_first_text":"Video's eerst","widget_review_pending_text":"In afwachting","social_share_options_order":"Facebook,Twitter,Pinterest,LinkedIn","preview_badge_show_question_text":true,"preview_badge_no_question_text":"0 Productvragen","preview_badge_n_question_text":"{{ number_of_questions }} vraag/vragen","qa_badge_show_icon":true,"widget_add_search_bar":true,"widget_search_bar_placeholder":"Doorzoek reviews","widget_sorting_verified_only_text":"Alleen geverifieerd","featured_carousel_theme":"aligned","featured_carousel_full_star_background":"#ffd600","all_reviews_page_load_more_text":"Laad meer recensies","widget_public_name_text":"openbaar weergegeven als","widget_reviewer_anonymous":"Anoniem","medals_widget_title":"Judge.me Review-medailles","platform":"woocommerce","branding_url":"https://judge.me/reviews","branding_text":"Via Judge.me","locale":"nl","reply_name":"Grill Bill Kamado BBQ","footer":true,"autopublish":true,"review_dates":true,"enable_custom_form":true};</script>
```

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description

Fix an issue with Judge.me that makes a JS script crash on load from a product page by excluding the script in question when Judge.me is loaded.

Fixes #4171

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which improves an existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Is the solution different from the one proposed during the grooming?
No grooming was right

Please describe in this section if there is any change to the solution, and why.

## How Has This Been Tested?

- [x] Automated testing
- [x] Test on my own web server

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by remyperona** [comment]
@wp-media/php I think we don't need a compatibility file here, we could directly add `window.jdgmSettings` to the exclusion array

**Comment by CrochetFeve0251** [comment]
@Tabrisrp should I wait someone confirm or should I refactor it now?

**Comment by remyperona** [comment]
You can refactor now

**Review by remyperona — CHANGES_REQUESTED** [review]
Cleanup in the composer file, bootstrap and remove the integration test

File: composer.json [hunk 1]
```diff
@@ -117,7 +117,7 @@
 	},
 	"scripts": {
 		"test-unit": "\"vendor/bin/phpunit\" --testsuite unit --colors=always --configuration tests/Unit/phpunit.xml.dist",
-		"test-integration": "\"vendor/bin/phpunit\" --testsuite integration --colors=always --configuration tests/Integration/phpunit.xml.dist --exclude-group AdminOnly,BeaverBuilder,Elementor,Hummingbird,WithSmush,WithWoo,WithAmp,WithAmpAndCloudflare,WithSCCSS,Cloudways,Dreampress,DoCloudflare,Multisite,WPEngine,SpinUpWP,WordPressCom,O2Switch,PDFEmbedder,PDFEmbedderPremium,PDFEmbedderSecure,Godaddy,LiteSpeed,RevolutionSlider,WordFence",
+		"test-integration": "\"vendor/bin/phpunit\" --testsuite integration --colors=always --configuration tests/Integration/phpunit.xml.dist --exclude-group AdminOnly,BeaverBuilder,Elementor,Hummingbird,WithSmush,WithWoo,WithAmp,WithAmpAndCloudflare,WithSCCSS,Cloudways,Dreampress,DoCloudflare,Multisite,WPEngine,SpinUpWP,WordPressCom,O2Switch,PDFEmbedder,PDFEmbedderPremium,PDFEmbedderSecure,Godaddy,LiteSpeed,RevolutionSlider,WordFence,JudgeMe",
```
Comment (remyperona) [inline composer.json]:
```suggestion
		"test-integration": "\"vendor/bin/phpunit\" --testsuite integration --colors=always --configuration tests/Integration/phpunit.xml.dist --exclude-group AdminOnly,BeaverBuilder,Elementor,Hummingbird,WithSmush,WithWoo,WithAmp,WithAmpAndCloudflare,WithSCCSS,Cloudways,Dreampress,DoCloudflare,Multisite,WPEngine,SpinUpWP,WordPressCom,O2Switch,PDFEmbedder,PDFEmbedderPremium,PDFEmbedderSecure,Godaddy,LiteSpeed,RevolutionSlider,WordFence",
```

Comment (CrochetFeve0251) [reply composer.json]: @CrochetFeve0251 I pushed 2 commits on that topic in parallel

**Review by remyperona — APPROVED** [review]

**Review by Mai-Saad — COMMENTED** [review]
Working fine now (note we will always exclude `window.jdgmSettings` even if the related plugin is not installed)

**Review by Mai-Saad — APPROVED** [review]

**Pull request merged**